pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1712,48 +1755,104 @@ END
-
-    SELECT d.* INTO STRICT dimension_row FROM _timescaledb_catalog.dimension d WHERE hypertable_id = hypertable_row.id ORDER BY id LIMIT 1;
-
-    SELECT _timescaledb_internal.time_to_internal(min_time) INTO STRICT min_time_internal;
-
-    FOR chunk_row IN
-        SELECT c.*
-        FROM _timescaledb_catalog.dimension_slice ds
-        INNER JOIN _timescaledb_catalog.chunk_constraint cc ON cc.dimension_slice_id = ds.id
-        INNER JOIN _timescaledb_catalog.chunk c ON cc.chunk_id = c.id
-        WHERE dimension_id = dimension_row.id
-        -- the range_starts are inclusive
-        AND min_time_internal >= ds.range_start
-        AND c.compressed_chunk_id IS NOT NULL
-        ORDER BY ds.range_start
-    LOOP
-
-        --lock the chunk exclusive.
-        EXECUTE format('LOCK %I.%I;', chunk_row.schema_name, chunk_row.table_name);
-        --double check it's still compressed.
-        PERFORM c.*
-        FROM _timescaledb_catalog.chunk c
-        WHERE c.id = chunk_row.id AND c.compressed_chunk_id IS NOT NULL;
+    -- in early versions of timescale multinode the access node catalog does not
+    -- store whether chunks were compressed, so we need to run the actual search
+    -- for nodes in need of decompression on the data nodes
+    IF SCHEMA_CATALOG.is_multinode() THEN
+        CALL distributed_exec(
+            format(
+                $dist$ CALL do_decompress_chunks_after(%s, %s) $dist$,
+                quote_literal(metric_table), quote_literal(min_time)),
+            transactional => false);
+    ELSE
+        CALL do_decompress_chunks_after(metric_table, min_time);
+    END IF;
+END
+$proc$ LANGUAGE PLPGSQL;
 
-        IF FOUND THEN
-            RAISE NOTICE 'Promscale is decompressing chunk: %.%', chunk_row.schema_name, chunk_row.table_name;
-            PERFORM decompress_chunk(format('%I.%I', chunk_row.schema_name, chunk_row.table_name)::regclass);
-        END IF;
+CALL execute_everywhere($ee$
+    CREATE OR REPLACE PROCEDURE SCHEMA_CATALOG.compress_old_chunks(metric_table TEXT, compress_before TIMESTAMPTZ)
+    AS $$
+    DECLARE
+        chunk TEXT;
+        chunk_num INT;
+    BEGIN
+        FOR chunk, chunk_num IN
+            SELECT
+                format('%I.%I', chunk_schema, chunk_name),
+                row_number() OVER (ORDER BY range_end DESC)
+            FROM timescaledb_information.chunks
+            WHERE hypertable_schema = 'SCHEMA_DATA'
+                AND hypertable_name = metric_table
+                AND NOT is_compressed
+                AND range_end <= compress_before
+            ORDER BY range_end ASC
+        LOOP
+            CONTINUE WHEN chunk_num <= 1;
+            PERFORM compress_chunk(chunk, if_not_compressed => true);
+            COMMIT;
+        END LOOP;
+    END;
+    $$ LANGUAGE PLPGSQL;
+$ee$);
 
-        COMMIT;
-    END LOOP;
-END;
-$$ LANGUAGE PLPGSQL;
+CREATE OR REPLACE PROCEDURE SCHEMA_CATALOG.compression_job(job_id int, config jsonb)
```

Comment:
let's be a bit more explicit in the naming so we don't clash with something later.
```suggestion
CREATE OR REPLACE PROCEDURE SCHEMA_CATALOG.promscale_compression_job(job_id int, config jsonb)
```

Reply:
remember, it's in `_prom_catalog` so the names are
`_prom_catalog.compression_job()` versus
`_prom_catalog. promscale_compression_job()`